src/ni/datastore/_auth.py

```diff
@@ -0,0 +1,442 @@
+        # and doesn't require spawning a subprocess.
+        #
+        # To implement the gRPC approach:
+        #
+        # 1. Add the nigel auth service proto package as a dependency in pyproject.toml
+        #    Example: ni-testhub-auth-v1-client = "^1.0.0"
+        #
+        # 2. Import and use the auth service stubs:
+        #    from ni.testhub.auth.v1 import auth_service_pb2, auth_service_pb2_grpc
+        #
+        # 3. Create the stub and make the gRPC call:
+        #    auth_daemon_address = self._get_auth_daemon_address()
+        #    channel = self._grpc_channel_pool.get_channel(auth_daemon_address)
+        #    stub = auth_service_pb2_grpc.AuthServiceStub(channel)
+        #    request = auth_service_pb2.GetTokenRequest()
+        #    response = stub.GetToken(request)
+        #    self._cached_token = response.token
+        #    return self._cached_token
+        #
+        # 4. Verify the provided_interface and service_class in _resolve_auth_daemon_via_discovery
+        #    match the actual values used by the nigel auth daemon
+        #
+        # For reference, see the daemon documentation at:
+        # https://github.com/ni/testhub/blob/main/src/cli/docs/daemon.md
+
+        raise RuntimeError(
+            "Unable to obtain JWT token. Tried:\n"
+            "1. NIGEL_SERVICES_JWT_TOKEN environment variable (not set)\n"
+            f"2. nigel CLI executable ({'found' if nigel_path else 'not found'})\n"
+            "3. gRPC auth daemon (not yet implemented)\n\n"
+            "To resolve:\n"
+            "- Set NIGEL_SERVICES_JWT_TOKEN environment variable with a valid token, or\n"
+            "- Install the nigel CLI and ensure it's in your PATH or set NIGEL_CLI_PATH, or\n"
+            "- Contact the development team to add gRPC proto definitions"
+        )
+
+    @staticmethod
+    def _find_nigel_executable() -> Optional[str]:
+        """Find the nigel CLI executable on the system.
+
+        Searches for the nigel executable in the following order:
+        1. NIGEL_CLI_PATH environment variable (if set)
+        2. System PATH
+        3. Platform-specific default installation paths
+
+        Returns:
+            The full path to the nigel executable, or None if not found.
+        """
+        # Check environment variable first
+        env_path = os.environ.get("NIGEL_CLI_PATH")
+        if env_path:
+            nigel_path = Path(env_path)
+            if nigel_path.exists() and nigel_path.is_file():
+                _logger.debug(f"Found nigel at NIGEL_CLI_PATH: {nigel_path}")
+                return str(nigel_path)
+
+        # Check if nigel is in the system PATH
+        nigel_in_path = shutil.which("nigel")
+        if nigel_in_path:
+            _logger.debug(f"Found nigel in PATH: {nigel_in_path}")
+            return nigel_in_path
+
+        # Platform-specific default paths
+        system = platform.system()
```

Comment:
I think I'd just get rid of all of this "platform specific default paths" stuff. It's not where we'll really install this thing - so lets not add the soon to be dead code.